packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -105,125 +103,123 @@ export class BlockSwitcher extends Component {
+					>
+						<div className="block-editor-block-switcher__preview">
+							<div className="block-editor-block-switcher__preview-title">
+								{ __( 'Preview' ) }
 							</div>
-						) }
-						{ hoveredClassName !== null && (
-							<div className="block-editor-block-switcher__preview">
-								<div className="block-editor-block-switcher__preview-title">
-									{ __( 'Preview' ) }
-								</div>
-								<BlockPreview
-									viewportWidth={ 500 }
-									blocks={
-										hoveredBlockType.example
-											? getBlockFromExample(
-													hoveredBlock.name,
-													{
-														attributes: {
-															...hoveredBlockType
-																.example
-																.attributes,
-															className: hoveredClassName,
-														},
-														innerBlocks:
-															hoveredBlockType
-																.example
-																.innerBlocks,
-													}
-											  )
-											: cloneBlock( hoveredBlock, {
-													className: hoveredClassName,
-											  } )
-									}
+							<BlockPreview
+								viewportWidth={ 500 }
+								blocks={
+									hoveredBlockType.example
+										? getBlockFromExample(
+												hoveredBlock.name,
+												{
+													attributes: {
+														...hoveredBlockType
+															.example.attributes,
+														className: hoveredClassName,
+													},
+													innerBlocks:
+														hoveredBlockType.example
+															.innerBlocks,
+												}
+										  )
+										: cloneBlock( hoveredBlock, {
+												className: hoveredClassName,
+										  } )
+								}
+							/>
+						</div>
+					</Popover>
+				</div>
+			</div>
+		);
+
+		const blockSwitcherLabel =
+			1 === blocks.length
+				? __( 'Change block type or style' )
+				: sprintf(
+						/* translators: %s: number of blocks. */
+						_n(
+							'Change type of %d block',
+							'Change type of %d blocks',
+							blocks.length
+						),
+						blocks.length
+				  );
+
+		return (
+			<ToolbarGroup>
+				<ToolbarItem>
+					{ ( toggleProps ) => (
+						<DropdownMenu
+							className="block-editor-block-switcher"
+							label={ blockSwitcherLabel }
+							popoverProps={ {
+								position: 'bottom right',
+								isAlternate: true,
+								className:
+									'block-editor-block-switcher__popover',
+							} }
+							icon={
+								<BlockIcon
+									icon={ icon }
+									className="block-editor-block-switcher__toggle"
+									showColors
 								/>
-							</div>
-						) }
-					</>
-				) }
-			/>
+							}
+							toggleProps={ toggleProps }
+							menuProps={ { orientation: 'both' } }
+						>
+							{ ( { onClose } ) =>
+								( hasBlockStyles ||
+									hasPossibleBlockTransformations ) && (
+									<div className="block-editor-block-switcher__container">
+										{ hoveredClassName !== null &&
+											PreviewBlockPopover }
+										{ hasBlockStyles && (
+											<MenuGroup>
```

Comment:
The same thing with the label here:
```suggestion
											<MenuGroup label={ __( 'Styles' ) }>
```